src/aolayer.cpp

```diff
@@ -277,6 +281,12 @@ void AOLayer::start_playback(QString p_image)
   if (!file_exists(p_image))
     return;
 
+  QString scaling_override = ao_app->read_design_ini("scaling", p_image + ".ini");
+  if (scaling_override == "smooth")
+    transform_mode = Qt::SmoothTransformation;
+  else if (scaling_override == "fast")
+    transform_mode = Qt::FastTransformation;
+
```

Comment:
Most people have upscaling in mind when they think about the filtering method, but they hardly consider how it will look when downscaling. You should not use FastTransformation when scaling down; otherwise the downscale will look horrible.

Also, there is now an ini file at a per-image basis? Oh, dear...

Reply:
Maybe the char.ini should be used per-emoite basis instead?

Reply:
scaling_override is intended as a last resort for when a more general-purpose configuration ini is not practical (i.e. custom_objections folders, speedlines inside character folders).

As for downscaling using Fast - this is rendered impossible by the final check done by set_pixmap() - which checks if the destination size is smaller than the source pixmap, and if it is, force smooth scaling.

Reply:
I'd like to see scaling overrides be defined in `char.ini` instead of `<image>.ini`, please:
```ini
[Scaling]
myemote = smooth
myemote2 = fast
```

Reply:
see https://github.com/AttorneyOnline/AO2-Client/pull/302/commits/45dbd308771e595fbfbb9c5381419ae3c9a14b33, though my implementation is per-file rather than per-emote for more flexibility:
```ini
[Scaling]
(a)myemote = smooth
(b)myemote = fast
anim/myemote = fast
```